extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/MicrometerConfigUnremovableProcessor.java

```diff
@@ -0,0 +1,17 @@
+package io.quarkus.micrometer.deployment;
+
+import io.quarkus.arc.deployment.UnremovableBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.micrometer.runtime.config.MicrometerConfig;
+
+public class MicrometerConfigUnremovableProcessor {
+
+    /**
+     * config objects are beans, but they are not unremovable by default
+     */
+    @BuildStep
+    UnremovableBeanBuildItem mpConfigAsBean() {
+        return UnremovableBeanBuildItem.beanTypes(MicrometerConfig.class);
+    }
```

Comment:
I think this build step can remain in the main Processor.. it was not originally tagged as "only if micrometer was enabled" .. but classes/beans that need the micrometer config are not going to be loaded if micrometer is disabled...

Reply:
That's what I thought initially, but then I looked at the git history, and this was added specifically to prevent an NPE when micrometer is disabled: https://github.com/quarkusio/quarkus/pull/25169 . Apparently `RestClientMetricsListener` is still around in that case and it will trigger loading of the `MicrometerConfig` class.

So, we should probably run this build step when Micrometer is disabled?

Reply:
why... oh... because it's a stupid service loader thing?

Some people love service loader, but I'm not a fan, for reasons like this. I can't just turn it off.. the config has to be preserved so the automagic thing discovered in the jar knows whether or not to do anything. :sigh: I wonder if there is a better way to do that?